zb-db/src/main/java/io/zeebe/db/TransactionContext.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.db;
+
+/** Represents the shared state of a database interaction */
+public interface TransactionContext {
+
+  /**
+   * Runs the commands like delete, put etc. in a transaction. Access of different column families
+   * inside this transaction are possible.
+   *
+   * <p>Reading key-value pairs via get or an iterator is also possible and will reflect changes,
+   * which are made during the transaction.
+   *
+   * <p><b>NOTE</b>: This will automatically commit the transaction and rollback on error
```

Comment:
Will this commit the transaction always? In the implementation, it only commits when it runs in a new transaction.

Reply:
You mean the nesting? Yeah this is true. I can update this comment this is just copied from the old code.

Reply:
Not nesting. The comment indicates that a call to `runInTransaction` always commit the transaction (or rollback). But that is not true in the implementation. Just updating the comment would be enough.